graylog2-server/src/main/java/org/graylog2/shared/journal/LocalKafkaJournal.java

```diff
@@ -711,15 +712,17 @@ public List<JournalReadEntry> read(long readOffset, long requestedMaximumCount,
                 lastOffset = messageAndOffset.offset();
 
                 final byte[] payloadBytes = ByteBufferUtils.readBytes(messageAndOffset.message().payload());
-                if (LOG.isTraceEnabled() || includeMessageId) {
+                if (includeMessageId) {
                     final byte[] keyBytes = ByteBufferUtils.readBytes(messageAndOffset.message().key());
                     messages.add(new JournalReadEntry(keyBytes, payloadBytes, messageAndOffset.offset()));
-                    LOG.trace("Read message {} contains {}", bytesToHex(keyBytes), bytesToHex(payloadBytes));
-                }
-                totalBytes += payloadBytes.length;
-                if (!includeMessageId) {
+                    if (LOG.isTraceEnabled()) {
```

Comment:
We are now only logging if `includeMessageID` is true, but I don't think trace logging is important enough to hold this up any longer, so I'm approving.